### PR TITLE
risk mfa mapping

### PR DESCRIPTION
**Pull request by pitbulk**
All those new methods may be tested before merging

### CODE REVIEW

File: src/main/java/com/onelogin/sdk/conn/Client.java
Comment (bzvestey) [inline src/main/java/com/onelogin/sdk/conn/Client.java]: The changes to the `Mapping` model means that this function fails to parse the response properly. Only the ID of the new mapping is return in an object like `{"id":763875634789}`. This is similar to how roles work.

File: src/main/java/com/onelogin/sdk/conn/Client.java
Comment (bzvestey) [inline src/main/java/com/onelogin/sdk/conn/Client.java]: Like create mappings, this function fails to parse the response properly. Only the ID of the updated mapping is return in an object like `{"id":763875634789}`.

File: src/main/java/com/onelogin/sdk/conn/Client.java
Comment (bzvestey) [inline src/main/java/com/onelogin/sdk/conn/Client.java]:
```suggestion
				info = jsonArray.getJSONObject(i);
				resultData = new HashMap<String, Object>();
				resultData.put("user", new User(info.getJSONObject("user")));
				resultData.put("mapped", info.getBoolean("mapped"));
```

Comment (bzvestey) [reply src/main/java/com/onelogin/sdk/conn/Client.java]: Some minor issues above that I saw while testing this.

File: src/main/java/com/onelogin/sdk/conn/Client.java
Comment (bzvestey) [inline src/main/java/com/onelogin/sdk/conn/Client.java]: From the error I was getting and what I can tell in the code `OneloginOAuthJSONResourceResponse` can't handle arrays, but `OneloginOAuth2JSONResourceResponse` seems to work instead. Some condition and action values can return an empty body if it can be random text, so dataArray can sometimes be null.

```suggestion
		OneloginOAuth2JSONResourceResponse oAuth2Response = (OneloginOAuth2JSONResourceResponse) executeCall(url, OAuth.HttpMethod.GET, OneloginOAuth2JSONResourceResponse.class);
		if (oAuth2Response.getResponseCode() == 200) {
			HashMap<String, String> info;
			JSONArray dataArray = oAuth2Response.getJSONArrayFromContent();
			if (dataArray == null) return null;
			
			for(int i = 0; i < dataArray.length(); i++) {
				JSONObject data = dataArray.getJSONObject(i);
				info = new HashMap<String, String>();
				info.put("name", data.optString("name"));
				info.put("value", data.optString("value"));
				listNameValue.add(info);
			}
		} else {
			setResponseError(oAuth2Response, false);
		}
```

File: src/main/java/com/onelogin/sdk/conn/Client.java
Comment (bzvestey) [inline src/main/java/com/onelogin/sdk/conn/Client.java:4104]: This fails, as the response is an array of ids, and not an array of objects with an id field. Not sure what the best option is fore changing this.

File: src/main/java/com/onelogin/sdk/conn/Client.java
Comment (bzvestey) [reply src/main/java/com/onelogin/sdk/conn/Client.java]: Not sure what the best option is for fixing this, as I don't know how reflection works in Java to see if it is possible to only match the id.

**Review by bzvestey — CHANGES_REQUESTED** [review]
So far I have only reviewed and tested the mappings related functionality.

File: src/main/java/com/onelogin/sdk/conn/Client.java
Comment (bzvestey) [inline src/main/java/com/onelogin/sdk/conn/Client.java]: From my testing the response from the API is `OK` on a 200 and is not a JSON object... not sure the best way to work with that here.

File: src/main/java/com/onelogin/sdk/conn/Client.java
Comment (pitbulk) [reply src/main/java/com/onelogin/sdk/conn/Client.java]: Ok so createResource should be used instead

https://github.com/onelogin/onelogin-java-sdk/blob/untested_risk_mfa_mapping/src/main/java/com/onelogin/sdk/conn/Client.java#L1560

As was implemented createRole

File: src/main/java/com/onelogin/sdk/conn/Client.java
Comment (pitbulk) [reply src/main/java/com/onelogin/sdk/conn/Client.java]: Here updateResource
https://github.com/onelogin/onelogin-java-sdk/blob/untested_risk_mfa_mapping/src/main/java/com/onelogin/sdk/conn/Client.java#L1612

File: src/main/java/com/onelogin/sdk/conn/Client.java
Comment (pitbulk) [reply src/main/java/com/onelogin/sdk/conn/Client.java]: :+1:

File: src/main/java/com/onelogin/sdk/conn/Client.java
Comment (pitbulk) [reply src/main/java/com/onelogin/sdk/conn/Client.java:4104]: Ideally extend the setOperation method to cover such use case as well

Check how was solved in python:
https://github.com/onelogin/onelogin-python-sdk/blob/v2/src/onelogin/api/client.py#L268


https://github.com/onelogin/onelogin-python-sdk/blob/v2/src/onelogin/api/util/response_handlers.py#L57

File: src/main/java/com/onelogin/sdk/conn/Client.java
Comment (pitbulk) [reply src/main/java/com/onelogin/sdk/conn/Client.java]: I see there is also a typo on the URL constant
SMART_MFA_VERFY_TOKEN
Should be
SMART_MFA_VERIFY_TOKEN

Comment (pitbulk) [reply src/main/java/com/onelogin/sdk/conn/Client.java]: In the code I tried to only check the header, as I expect to get other status code if it fails.

But I agree this is risky. The endpoint should return a valid formated result and not just a OK.

Maybe in order to improve security, avoiding verify_token to retur 
200 on calls to wrong urls that could return a simply 200,
maybe you could as well verify that the body length of the response is short as well as its OK.

Comment (pitbulk) [reply src/main/java/com/onelogin/sdk/conn/Client.java]: Python SDK should be improved as well
https://github.com/onelogin/onelogin-python-sdk/blob/v2/src/onelogin/api/client.py#L3222

File: src/main/java/com/onelogin/sdk/conn/Client.java
Comment (bzvestey) [inline src/main/java/com/onelogin/sdk/conn/Client.java:2470]: This fails, as createResource for API version 2 only checks for 201 response code. Is it ok to check for a 200 response code here also?

**Review by bzvestey — CHANGES_REQUESTED** [review]
@pitbulk I have some questions on some issues I found that I would like guidance on how to fix. Those that seemed simple enough, I will push up fixes for those.

File: src/main/java/com/onelogin/sdk/conn/Client.java
Comment (bzvestey) [reply src/main/java/com/onelogin/sdk/conn/Client.java]: I have a fix in place for dealing with the getting a string of `OK` back, basically a new response type that only parses the body as JSON on an error.

**Comment by bzvestey** [comment]
@pitbulk I have updated the branch with what I feel are the correct fixed, please have a look and let me know what you think.

File: src/main/java/com/onelogin/sdk/conn/Client.java
Comment (bzvestey) [reply src/main/java/com/onelogin/sdk/conn/Client.java:4104]: Updated to handle the fact that responses might not be JSON objects for getting the ID.

**Comment by bzvestey** [comment]
@eriktalvi feel free to take a look over this and let us know if you have any thoughts.

File: src/main/java/com/onelogin/sdk/conn/Client.java
Comment (pitbulk) [reply src/main/java/com/onelogin/sdk/conn/Client.java:2470]: Yes, let's check for 201 and 200

Comment (bzvestey) [reply src/main/java/com/onelogin/sdk/conn/Client.java:2470]: Cool, I have done that.

**Comment by pitbulk** [comment]
Looks good

**Review by bzvestey — APPROVED** [review]
As I have taken over this PR, I will wait for @eriktalvi to approve this before merging.

File: src/main/java/com/onelogin/sdk/model/FactorEnrollmentResponse.java
Comment (eriktalvi) [inline src/main/java/com/onelogin/sdk/model/FactorEnrollmentResponse.java:52]: get not ge

File: src/main/java/com/onelogin/sdk/model/Rule.java
Comment (eriktalvi) [inline src/main/java/com/onelogin/sdk/model/Rule.java]: only place with single line ifs. not huge issue but is breaking convention worth saving 5 lines?

**Review by eriktalvi — CHANGES_REQUESTED** [review]
The `getId` change is necessary

File: src/main/java/com/onelogin/sdk/model/FactorEnrollmentResponse.java
Comment (eriktalvi) [reply src/main/java/com/onelogin/sdk/model/FactorEnrollmentResponse.java:52]: its the line below .. oops

Comment (eriktalvi) [reply src/main/java/com/onelogin/sdk/model/FactorEnrollmentResponse.java:52]: its the line below.. Oops

`	public String geID() {`

**Review by eriktalvi — APPROVED** [review]

**Pull request merged**
